service_configuration_lib/spark_config.py

```diff
@@ -647,6 +647,10 @@ def _calculate_resources(
         'spark.executor.instances': str(executor_instances),
         'spark.task.cpus': str(task_cpus),
     })
+    if 'spark.cores.max' in user_spark_opts:
```

Comment:
I think if `spark.cores.max` is not in user_spark_opts, we should still set it, right?
That way users can remove this.

Reply:
I think the reason why we cannot just remove the config is because the `spark.executor.instances` is calculated from `spark.cores.max` (before this function) if `spark.executor.instances` is not specified. We would tend to be not adding it if it's not required.

Reply:
> I think the reason why we cannot just remove the config is because the spark.executor.instances is calculated from spark.cores.max ..
Agree.

Should we still set the config, `spark.cores.max` internally in our code, if not set by user?

Reason case:
User is giving both `spark.cores.max` and `spark.executor.instances`. It's common in on-point queries, to ask user to remove `spark.cores.max` config or replace with `spark.executor.instances`. This is done to avoid confusion of inconsistent values of (`spark.executor.instances` * `spark.instances.cores`) and `spark.cores.max` or even as a standard practice. 
Sometimes users can still set use this config in their code leading to failure of job or inconsistent with `spark.executor.instances * spark.executor.cores`

Having this set internally (if not present), would give us freedom to remove this config in spark configs:
* without breaking code
* or without risk of inconsistent expectations 

Thoughts?

Reply:
Ok, in that case, no strong preferences. 
Feel free to ignore this comment if doesn't looks useful to address in this case.